File: backend/routes/player.js
import express from 'express';
import { Player, DailyLog, Quest, Skill } from '../models/models.js';

const router = express.Router();

router.get('/player', async (req, res) => {
  try {
    let player = await Player.findOne();
    if (!player) {
      return res.status(404).json({ detail: "Player not found" });
    }

    // Check if we need to create logs for today
    const today = new Date().toISOString().split('T')[0];
    const existingLogs = await DailyLog.findAll({ where: { date: today } });
    
    if (existingLogs.length === 0) {
      const activeQuests = await Quest.findAll({ where: { is_active: true, is_daily: true } });
      const newLogs = activeQuests.map(q => ({
        date: today,
        quest_id: q.id,
        completed: false
      }));
      await DailyLog.bulkCreate(newLogs);
    }

    res.json(player);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// Get player skills
router.get('/player/skills', async (req, res) => {
  try {
    const skills = await Skill.findAll();
    res.json(skills);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// Add free points to a specific stat
router.post('/player/allocate-points', async (req, res) => {
  try {
    const { stat, amount } = req.body;
    const validStats = ['strength', 'agility', 'vitality', 'intelligence', 'sense'];
    
    if (!validStats.includes(stat)) {
      return res.status(400).json({ detail: "Invalid stat" });
    }
    
    const player = await Player.findOne();
    if (!player || player.free_points < amount) {
      return res.status(400).json({ detail: "Not enough free points" });
    }
    
    player[stat] += amount;
    player.free_points -= amount;
    await player.save();
    
    res.json({ message: "Points allocated", player });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

export default router;

File: backend_py_old/routes/player.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from ..database import get_db
from ..models import Player, DailyLog, Quest
from ..schemas import Player as PlayerSchema
from datetime import date

router = APIRouter()

@router.get("/player", response_model=PlayerSchema)
def get_player(db: Session = Depends(get_db)):
    player = db.query(Player).first()
    if not player:
        raise HTTPException(status_code=404, detail="Player not found")
    
    # Check if we need to create logs for today
    today = date.today()
    existing_logs = db.query(DailyLog).filter(DailyLog.date == today).all()
    if not existing_logs:
        quests = db.query(Quest).filter(Quest.is_active == True).all()
        for q in quests:
            new_log = DailyLog(date=today, quest_id=q.id, completed=False)
            db.add(new_log)
        db.commit()
        db.refresh(player)

    return player
